test/e2e/sync_waves_test.go

```diff
@@ -36,17 +36,6 @@ func TestFixingDegradedApp(t *testing.T) {
 		When().
 		Sync().
 		Then().
-		Expect(OperationPhaseIs(OperationFailed)).
-		Expect(SyncStatusIs(SyncStatusCodeOutOfSync)).
-		Expect(HealthIs(health.HealthStatusMissing)).
-		Expect(ResourceResultNumbering(1)).
-		Expect(ResourceSyncStatusIs("Pod", "pod-1", SyncStatusCodeSynced)).
-		Expect(ResourceHealthIs("Pod", "pod-1", health.HealthStatusHealthy)).
-		Expect(ResourceSyncStatusIs("Pod", "pod-2", SyncStatusCodeOutOfSync)).
-		Expect(ResourceHealthIs("Pod", "pod-2", health.HealthStatusMissing)).
-		When().
-		Sync().
-		Then().
```

Comment:
The reason this was deleted is because this test had two syncs in a row (even though nothing changed in between those two syncs). The first sync would immediately detect the pod in a ImagePullBackoff state and always fail.

After my change, the 2-second delay actually allowed the pod time to go from Degraded->Healthy, so that by the time the controller re-reconciled the operation, it saw the pod was healthy and the sync would succeed.